pkg/querier/queryrange/marshal.go

```diff
@@ -121,7 +121,9 @@ func ResultToResponse(result logqlmodel.Result, params logql.Params) (queryrange
 		sk, err := data.ToProto()
 		return &TopKSketchesResponse{Response: sk}, err
 	case logql.ProbabilisticQuantileMatrix:
-		return &QuantileSketchResponse{Response: data.ToProto()}, nil
+		r := data.ToProto()
+		data.Release()
```

Comment:
So, the GC would do this anyways. I'm skeptical that we really do gain some memory by doing this. That said we are still faster because we won't have to allocate as often in `JoinQuantileSketchVector`.

Reply:
It depends a bit on parallelization.

The more queries we execute in parallel when using the pool the more memory we're going to allocate, but with less parallel queries running will allocate less. Previously we always would have allocated even more, essentially always worst case scenario, since we allocated a slice on every call to `JoinQuantileSketchVector`.